src/fuelup_cli.rs

```diff
@@ -46,6 +51,41 @@ enum Commands {
 pub fn fuelup_cli() -> Result<()> {
     let cli = Cli::parse();
 
+    if let Some(toolchain_override) = ToolchainOverride::from_project_root() {
+        let override_path = toolchain_override.cfg.toolchain.channel.to_string();
+        let toolchain = match DistToolchainDescription::from_str(&override_path) {
+            Ok(desc) => Toolchain::from_path(&desc.to_string()),
+            Err(_) => Toolchain::from_path(&override_path),
+        };
+
+        info!("Using override toolchain '{}'", &toolchain.name);
+
+        if !toolchain.exists() {
+            match cli.command {
+                Commands::Toolchain(_) => {
```

Comment:
We only want to do this for operations where the toolchain override will be used in some way, right? So we also don't need to do this for `fuelup update/upgrade/install` `fuelup completions` `fuelup check` `fuelup default`

We really only need it for proxy commands and for `fuelup component` which modifies the currently active toolchain.

For `fuelup show` it could be good to show the current state first and then prompt the user if they wish to install.